src/GraphQL.ApiTests/GraphQL.approved.txt

```diff
@@ -2507,11 +2507,19 @@ namespace GraphQL.Utilities
         public TType GetMetadata<TType>(string key, TType defaultValue = default) { }
         public bool HasMetadata(string key) { }
     }
+    public enum NameType
```

Comment:
Will there be confusion with `NamedType`?

Reply:
I didn't think of that.  If you have a suggestion, I'm all ears.  I couldn't think of a good name.

Reply:
If you would rather skip this PR, that's fine too, as it shouldn't really matter.  But in principle I think an enum is the right choice for the method.

Reply:
50/50

Reply:
I like the use of the enum.  👍🏼